files/image_config/corefile_uploader/core_uploader.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python
+
+import os
+import time
+import tarfile
+import socket
+import yaml
+import json
+import syslog
+from shutil import rmtree
+from watchdog.observers import Observer
+from watchdog.events import FileSystemEventHandler
+from azure.storage.file import FileService
+
+global CORE_FILE_PATH, RC_FILE
+global hostname, sonicversion, asicname, acctname, acctkey, sharename, cwd
+global INIT_CWD
+global log_level
+global this_file
+
+this_file = os.path.basename(__file__)
+
+global cfg
+cfg = ""
+
+CORE_FILE_PATH = "/var/core/"
+RC_FILE = "/etc/sonic/core_analyzer.rc.json"
+INIT_CWD = "/tmp"
+
+hostname = ""
+sonicversion = ""
+asicname = ""
+acctname = ""
+acctkey = ""
+sharename = ""
+cwd = []
+
+HOURS_4 = (4 * 60 * 60)
+PAUSE_ON_FAIL = (60 * 60)
+MAX_RETRIES = 5
+
+log_level = syslog.LOG_DEBUG
+
+def log_msg(lvl, fname, m):
+    if (lvl <= log_level):
+        syslog.syslog(lvl, "{}: {}".format(fname, m))
+
+    if log_level == syslog.LOG_DEBUG:
+        print("{}: {}".format(fname, m))
+
+def log_err(m):
+    log_msg(syslog.LOG_ERR, this_file, m)
+
+def log_info(m):
+    log_msg(syslog.LOG_INFO, this_file, m)
+
+def log_warn(m):
+    log_msg(syslog.LOG_WARNING, this_file, m)
+
+def log_debug(m):
+    log_msg(syslog.LOG_DEBUG, this_file, m)
+
+
+def mkdir(p):
+    l = p.split("/")
+    if p.endswith("/"): 
+        l.pop()
+
+    e = l[0:len(l)]
+
+    while not os.path.exists("/".join(e)):
+        e.pop()
+
+    while len(e) != len(l):
+        e.append(l[len(e)])
+        os.mkdir("/".join(e))
```

Comment:
You can call shell command 'mkdir -p <full path>'

Reply:
I could. Let me.

Reply:
Done.